buildSrc/src/main/kotlin/buildsrc/tasks/ConvertCommonTestResourcesToKotlin.kt

```diff
@@ -0,0 +1,90 @@
+package buildsrc.tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.FileSystemOperations
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import java.nio.file.Path
+import javax.inject.Inject
+
+/**
+ * Generate Kotlin code that contains the test resources from the common source set.
+ *
+ * This is necessary for accessing test data in Kotlin Multiplatform tests, because there's
+ * currently no easy way to access and traverse filesystem content in Kotlin Multiplatform code.
+ */
+@CacheableTask
+abstract class ConvertCommonTestResourcesToKotlin @Inject constructor(
+    private val fs: FileSystemOperations,
+) : DefaultTask() {
+    /** The directory that will contain the generated Kotlin code. */
+    @get:OutputDirectory
+    abstract val destination: DirectoryProperty
+
+    // HACK: Ideally we'd retrieve the 'resources' path via Gradle API, but somehow Gradle sees only 'main' and 'test'
+    // source sets instead of the required 'commonTest'.
+    private val commonResourcesDir: File = project.rootProject.projectDir
```

Comment:
issue: if I am not mistaken, we should use `project.projectDir`. Right now the task always refers to the root project

Reply:
Right, fixed!

Reply:
https://github.com/krzema12/snakeyaml-engine-kmp/pull/312/commits/46eb3c33547738b207299af227f8b39014780c32